gammapy/obs/observation.py

```diff
@@ -85,3 +91,244 @@ def select_linspace_subset(self, num):
         # Round down to nearest integer
         indices = indices.astype('int')
         return self[indices]
+
+    def filter_obs_var_range(self, selection_variable,
+                             value_min, value_max, inverted=False):
+        """Make an observation table, applying some selection.
+
+        Generic function to apply a 1D box selection (min, max) to a
+        table on any variable that is in the observation table and can
+        be casted into a `~astropy.units.Quantity` object.
+
+        If the inverted flag is activated, the filter is applied to
+        keep all elements outside the selected range.
+
+        Parameters
+        ----------
+        selection_variable : string
+            name of variable to apply a cut (it should exist on the table)
+        value_min : `~astropy.units.Quantity`-like
+            minimum value; type should be consistent with selection_variable
+        value_max : `~astropy.units.Quantity`-like
+            maximum value; type should be consistent with selection_variable
+        inverted : bool, optional
+            invert selection: keep all entries outside the (min, max) range
+
+        Returns
+        -------
+        obs_table : `~gammapy.obs.ObservationTable`
+            observation table with observations passing the selection
+        """
+        obs_table = self
+
+        # check that variable exists on the table
+        if selection_variable not in obs_table.keys():
+            raise KeyError('Key not present in table: {}'.format(selection_variable))
+
+        # read values into a quantity in case units have to be taken into account
+        value = Quantity(obs_table[selection_variable])
+
+        # build and apply mask
+        if not inverted:
+            mask = (value_min < value) & (value < value_max)
+        else:
+            mask = (value_min >= value) | (value >= value_max)
```

Comment:
Instead of a separate expression for the inverse mask, I think this is a bit easier:
```
mask = (value_min < value) & (value < value_max)

if inverted:
    mask = np.invert(mask)
```
Use the same pattern for the other functions where you have an `inverted` option.